pkg/graphql/schema.go

```diff
@@ -119,6 +132,179 @@ func (s *Schema) IntrospectionResponse(out io.Writer) error {
+			if skip {
+				continue
+			}
+
+			s.addTypeFieldArgsForFieldRef(fieldRef, typeName, fieldName, &typeFieldArguments)
+		}
+	}
+
+	return typeFieldArguments
+}
+
+func (s *Schema) typeNameOfObjectTypeIfHavingFields(objectType ast.ObjectTypeDefinition) (typeName string, ok bool) {
+	if !objectType.HasFieldDefinitions {
+		return "", false
+	}
+
+	return string(s.document.Input.ByteSlice(objectType.Name)), true
+}
+
+func (s *Schema) fieldNameOfFieldDefinitionIfHavingArguments(field ast.FieldDefinition, ref int) (fieldName string, ok bool) {
+	if !field.HasArgumentsDefinitions {
+		return "", false
+	}
+
+	return s.document.FieldDefinitionNameString(ref), true
+}
+
+func (s *Schema) determineIfFieldWithFieldNameShouldBeSkipped(ref int, typeName string, skipFieldFuncs ...SkipFieldFunc) (fieldName string, skip bool) {
+	field := s.document.FieldDefinitions[ref]
+	fieldName, ok := s.fieldNameOfFieldDefinitionIfHavingArguments(field, ref)
+	if !ok {
+		return fieldName, true
+	}
+
+	for _, skipFieldFunc := range skipFieldFuncs {
+		if skipFieldFunc != nil && skipFieldFunc(typeName, fieldName, s.document) {
+			skip = true
+			break
+		}
+	}
+
+	return fieldName, skip
+}
+
+func (s *Schema) addTypeFieldArgsForFieldRef(ref int, typeName string, fieldName string, fieldArguments *[]TypeFieldArguments) {
+	currentTypeFieldArgs := TypeFieldArguments{
+		TypeName:      typeName,
+		FieldName:     fieldName,
+		ArgumentNames: make([]string, 0),
+	}
+
+	for _, argRef := range s.document.FieldDefinitions[ref].ArgumentsDefinition.Refs {
+		argName := s.document.Input.ByteSlice(s.document.InputValueDefinitions[argRef].Name)
+		currentTypeFieldArgs.ArgumentNames = append(currentTypeFieldArgs.ArgumentNames, string(argName))
+	}
+
+	*fieldArguments = append(*fieldArguments, currentTypeFieldArgs)
+}
+
+func (s *Schema) GetAllNestedFieldChildrenFromTypeField(typeName string, fieldName string, skipFieldFuncs ...SkipFieldFunc) []TypeFields {
+	fields := s.nodeFieldRefs(typeName)
+	if len(fields) == 0 {
+		return nil
+	}
+	for _, ref := range fields {
+		if fieldName == s.document.FieldDefinitionNameString(ref) {
+			fieldTypeName := s.document.FieldDefinitionTypeNode(ref).NameString(&s.document)
+			childNodes := make([]TypeFields, 0)
+			s.findNestedFieldChildren(fieldTypeName, &childNodes, skipFieldFuncs...)
+			return childNodes
+		}
+	}
+
+	return nil
+}
+
+func (s *Schema) findNestedFieldChildren(typeName string, childNodes *[]TypeFields, skipFieldFuncs ...SkipFieldFunc) {
+	fields := s.nodeFieldRefs(typeName)
+	if len(fields) == 0 {
+		return
+	}
+	for _, ref := range fields {
+		fieldName := s.document.FieldDefinitionNameString(ref)
+		if len(skipFieldFuncs) > 0 {
+			skip := false
+			for _, skipFieldFunc := range skipFieldFuncs {
+				if skipFieldFunc != nil && skipFieldFunc(typeName, fieldName, s.document) {
+					skip = true
+					break
+				}
+			}
+
+			if skip {
+				continue
+			}
+		}
+		s.putChildNode(childNodes, typeName, fieldName)
```

Comment:
I believe we could still have a recursive crash here as we should not run `findNestedFieldChildren` when child node was already added

please have a look at this PR: https://github.com/TykTechnologies/tyk/pull/3471
it could be discarded in case u will add the same test for countries schema and similar changes here